### PR TITLE
ASimNetworkSessionBarracudaWAF.yaml-33

### PR DESCRIPTION
**Pull request by Alekhya0824**
Required items, please complete
   
   Change(s):
   - Created a new file for BaracudeCEF and added BaracudaCEF parser in it. In BaracudaWAF file removed BaracudaCEF parser and in ASimNetworkSession added BaracudeCEF Parser name  and same changes have done for vim as well.

   Reason for Change(s):
   - To Split the BaracudaWAF and BaracudaCEF parser separately

   Version Updated:
   - yes

   Testing Completed:
   - yes

   Checked that the validations are passing and have addressed any issues that are present:
   - yes

### CODE REVIEW

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
KQL validation failures, please check for those. Thanks
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Validations failing please check. Thanks

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+    Type: dynamic
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let barracudaSchema = datatable(
```
Comment (vakohl) [inline Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml:53]: we dont need this schema for CEF parser

**Review by vakohl — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -335,4 +335,5 @@ ASALocalRun/
 Hunting Queries/DeployedQueries.json
 
 #Mac OSx
-.DS_Store
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply .gitignore:338]: after updating schema...it's considering old schema file so to consider new schema update added this line

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml (hunk 1 above)
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml:53]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml:1]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml [hunk 4]
```diff
@@ -0,0 +1,229 @@
+Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml:1]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaWAF.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaWAF.yaml:1]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaCEF.yaml [hunk 6]
```diff
@@ -0,0 +1,157 @@
+Parser:
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaCEF.yaml:1]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml [hunk 7]
```diff
@@ -15,7 +15,7 @@ References:
 - Title: Logs type Description in Barracuda WAF
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml:15]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml [hunk 8]
```diff
@@ -1,7 +1,7 @@
 Parser:
   Title: Network Session ASIM filtering parser for Barracuda WAF
-  Version: '0.1'
-  LastUpdated: Jun 13 2023
+  Version: '0.2.1'
+  LastUpdated: Jun 5 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml:4]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+Product:
+  Name: Barracuda WAF
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml:14]: done

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml [hunk 10]
```diff
@@ -0,0 +1,229 @@
+Parser:
+  Title: Network Session ASIM filtering parser for Barracuda WAF
+  Version: '0.2.1'
+  LastUpdated: JunE 5 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaCEF.yaml:4]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaWAF.yaml [hunk 11]
```diff
@@ -1,7 +1,7 @@
...
   Title: Network Session ASIM parser for Barracuda WAF
-  Version: '0.1'
-  LastUpdated: Jun 13 2023
+  Version: '0.2.1'
+  LastUpdated: JunE 5 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaWAF.yaml:4]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaCEF.yaml [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+Product:
+  Name: Barracuda WAF
+Normalization:
+  Schema: NetworkSession
+  Version: '0.2.6'
+References:
+- Title: ASIM Network Session Schema
+  Link: https://aka.ms/ASimNetworkSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaCEF.yaml:14]: done

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaCEF.yaml [hunk 13]
```diff
@@ -0,0 +1,157 @@
+Parser:
+  Title: Network Session ASIM parser for Barracuda WAF
+  Version: '0.2.1'
+  LastUpdated: June 5 2024
```
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaCEF.yaml:4]: done

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
I can see a few open comments from the reviewer, and PR is also having failed validations. Please act on them. 
image

And also, one suggestion: once you're done with the open comment change, please click on the `Resolve Conversation` button.
image

**Comment by v-atulyadav** [comment]
Hi @Alekhya0824,
Please act on above comments and also look for validations that failed. Thanks

File: Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaWAF.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (Alekhya0824) [reply Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionBarracudaWAF.yaml:1]: As data is not there in workspace so closing this comment

File: Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (vakohl) [reply Parsers/ASimNetworkSession/Parsers/vimNetworkSessionBarracudaWAF.yaml:1]: please un-comment
image

**Review by vakohl — APPROVED** [review]

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**